Mathlib/RingTheory/Binomial.lean

```diff
@@ -254,13 +254,13 @@ instance Int.instBinomialRing : BinomialRing ℤ where
         ← Int.neg_ofNat_succ, ascPochhammer_smeval_neg_eq_descPochhammer]
 
 noncomputable instance {R : Type*} [AddCommMonoid R] [Module ℚ≥0 R] [Pow R ℕ] : BinomialRing R where
-  nsmul_right_injective n hn r s hrs := by
+  nsmul_right_injective {n} hn r s hrs := by
     rw [← one_smul ℚ≥0 r, ← one_smul ℚ≥0 s, show 1 = (n : ℚ≥0)⁻¹ • (n : ℚ≥0) by simp_all]
     simp_all only [smul_assoc, Nat.cast_smul_eq_nsmul]
   multichoose r n := (n.factorial : ℚ≥0)⁻¹ • Polynomial.smeval (ascPochhammer ℕ n) r
   factorial_nsmul_multichoose r n := by
-    simp only [← smul_assoc]
-    field_simp
+    rw [← smul_assoc, nsmul_eq_mul, mul_inv_cancel₀ (Nat.cast_ne_zero.mpr n.factorial_ne_zero),
```

Comment:
What's the motivation for this change? Is it faster?
(In general, I believe un-automating is not wanted - unless it results in a big speed-up, or similar.)

Reply:
Okay, I will undo the change.